app/services/agency_identity_linker.rb

```diff
@@ -0,0 +1,49 @@
+class AgencyIdentityLinker
+  def initialize(sp_identity)
+    @sp_identity = sp_identity
+    @agency_id = nil
+  end
+
+  def link_identity
+    ai = agency_identity_from_sp_identity
```

Comment:
We typically prefer long form names over short form ones like this so the code can stay readable. Thought on changing this to `agency_identity`.

Reply:
I like that idea.